optuna/multi_objective/_hypervolume/wfg.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+
+from optuna.multi_objective._hypervolume import _compute_2points_volume
+from optuna.multi_objective._hypervolume import BaseHypervolume
+
+
+class WFG(BaseHypervolume):
+    """Hypervolume calculator for any dimension.
+
+    This class exactly calculates the hypervolume for any dimension by using the WFG algorithm.
+    For detail, see `While, Lyndon, Lucas Bradstreet, and Luigi Barone. "A fast way of
+    calculating exact hypervolumes." Evolutionary Computation, IEEE Transactions on 16.1 (2012)
+    : 86-95.`.
+    """
+
+    def __init__(self) -> None:
+        self._reference_point = None
+
+    def compute(self, solution_set: np.ndarray, reference_point: np.ndarray) -> float:
+        self._reference_point = reference_point
```

Comment:
We you going to validate the arguments that the reference point is dominated by all points?